0081-combine-and-effects-pt2/ReactiveSwiftPrimeTime/FavoritePrimes/FavoritePrimes.swift

```diff
@@ -0,0 +1,86 @@
+import ComposableArchitecture
+import SwiftUI
+
+public enum FavoritePrimesAction {
+  case deleteFavoritePrimes(IndexSet)
+  case loadButtonTapped
+  case loadedFavoritePrimes([Int])
+  case saveButtonTapped
+}
+
+public func favoritePrimesReducer(state: inout [Int], action: FavoritePrimesAction) -> [Effect<FavoritePrimesAction>] {
+  switch action {
+  case let .deleteFavoritePrimes(indexSet):
+    for index in indexSet {
+      state.remove(at: index)
+    }
+    return []
+
+  case let .loadedFavoritePrimes(favoritePrimes):
+    state = favoritePrimes
+    return []
+
+  case .saveButtonTapped:
+    return [saveEffect(favoritePrimes: state)]
+
+  case .loadButtonTapped:
+    return [
+        loadEffect
+            .skipNil()
+    ]
+    }
+}
+
+private func saveEffect(favoritePrimes: [Int]) -> Effect<FavoritePrimesAction> {
+  return .fireAndForget {
+    let data = try! JSONEncoder().encode(favoritePrimes)
+    let documentsPath = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0]
+    let documentsUrl = URL(fileURLWithPath: documentsPath)
+    let favoritePrimesUrl = documentsUrl.appendingPathComponent("favorite-primes.json")
+    try! data.write(to: favoritePrimesUrl)
+  }
+}
+
+import Combine
+
+private let loadEffect = Effect<FavoritePrimesAction?> { obs, _ in
+  let documentsPath = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0]
+  let documentsUrl = URL(fileURLWithPath: documentsPath)
+  let favoritePrimesUrl = documentsUrl.appendingPathComponent("favorite-primes.json")
+  guard
+    let data = try? Data(contentsOf: favoritePrimesUrl),
+    let favoritePrimes = try? JSONDecoder().decode([Int].self, from: data)
+    else { return  }
+    obs.send(value: .loadedFavoritePrimes(favoritePrimes))
+    obs.sendCompleted()
```

Comment:
I believe you could create a `sync` that simply works `work` and then immediately completes. That way the ReactiveSwift would match exactly the API we have for combine (and the other Rx branch). Could you give that a shot?

Reply:
Yeah, not that hard, fixed.